File: bootcamp-ai/src/project-alpha/backend/app/schemas/ticket.py
from pydantic import BaseModel, Field
from datetime import datetime
from uuid import UUID
from typing import Optional, List
from app.models.ticket import TicketPriority, TicketStatus
from app.schemas.label import LabelBase


class TicketCreate(BaseModel):
    """创建 Ticket 请求模型"""
    title: str = Field(..., min_length=1, max_length=200)
    description: Optional[str] = Field(None, max_length=10000)
    priority: TicketPriority = TicketPriority.MEDIUM
    label_ids: List[UUID] = Field(default_factory=list)


class TicketUpdate(BaseModel):
    """更新 Ticket 请求模型"""
    title: Optional[str] = Field(None, min_length=1, max_length=200)
    description: Optional[str] = Field(None, max_length=10000)
    priority: Optional[TicketPriority] = None
    label_ids: Optional[List[UUID]] = None


class TicketResponse(BaseModel):
    """Ticket 响应模型"""
    id: UUID
    title: str
    description: Optional[str] = None
    priority: TicketPriority
    status: TicketStatus
    labels: List[LabelBase]
    created_at: datetime
    updated_at: datetime
    completed_at: Optional[datetime] = None
    
    class Config:
        from_attributes = True


class TicketStatusResponse(BaseModel):
    """Ticket 状态更新响应模型"""
    id: UUID
    status: TicketStatus
    completed_at: Optional[datetime] = None

File: bootcamp-ai/src/project-alpha/frontend/src/types/ticket.ts
export const TICKET_PRIORITIES = ['low', 'medium', 'high', 'critical'] as const;
export const TICKET_STATUSES = ['open', 'completed'] as const;

export type TicketPriority = typeof TICKET_PRIORITIES[number];
export type TicketStatus = typeof TICKET_STATUSES[number];

export interface Label {
  id: string;
  name: string;
  color: string;
  description?: string | null;
}

export interface Ticket {
  id: string;
  title: string;
  description?: string | null;
  priority: TicketPriority;
  status: TicketStatus;
  labels: Label[];
  created_at: string;
  updated_at: string;
  completed_at?: string | null;
}

export interface CreateTicketInput {
  title: string;
  description?: string;
  priority?: TicketPriority;
  label_ids?: string[];
}

export interface UpdateTicketInput {
  title?: string;
  description?: string;
  priority?: TicketPriority;
  label_ids?: string[];
}

export interface TicketFilters {
  title?: string;
  status?: 'all' | TicketStatus;
  priority?: TicketPriority;
  label_ids?: string[];
  no_label?: boolean;
  sort_by?: 'created_at' | 'updated_at' | 'priority';
  sort_order?: 'asc' | 'desc';
  page?: number;
  page_size?: number;
}

export interface PaginationMeta {
  page: number;
  page_size: number;
  total: number;
  total_pages: number;
  has_next: boolean;
  has_prev: boolean;
}

export interface PaginatedTicketsResponse {
  data: Ticket[];
  pagination: PaginationMeta;
}
